websocket/polygon_test.go

```diff
@@ -1,99 +1,57 @@
-package polygonws_test
+package polygonws
 
 import (
-	"context"
-	"encoding/json"
-	"fmt"
-	"os"
 	"testing"
-	"time"
 
-	polygonws "github.com/polygon-io/client-go/websocket"
 	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
 )
 
-func TestMain(t *testing.T) {
-	apiKey := os.Getenv("API_KEY")
-	if apiKey == "" {
-		return // skip in CI for now
-	}
-
-	ctx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
-	defer cancel()
-
-	log := logrus.New()
-	log.SetLevel(logrus.DebugLevel)
-	c, err := polygonws.New(polygonws.Config{
-		APIKey:    apiKey,
-		Feed:      polygonws.RealTime,
-		Market:    polygonws.Stocks,
-		ParseData: true, // comment for raw data handling
-		Log:       log,
-	})
-	if err != nil {
-		log.Fatal(err)
-	}
-	defer c.Close()
-
-	c.Close() // this shouldn't panic
-	if err := c.Connect(); err != nil {
-		log.Fatal(err)
-	}
-
-	// calling connect again shouldn't panic or data race
-	if err := c.Connect(); err != nil {
-		log.Fatal(err)
-	}
-
-	go printOutput(c) // comment for raw data handling
-	// go printRawOutput(c) // uncomment for raw data handling
-
-	if err := c.Subscribe(polygonws.StocksSecAggs, "AAPL", "MSFT"); err != nil {
-		log.Error(err)
-	}
-
-	time.Sleep(5 * time.Second)
-	if err := c.Subscribe(polygonws.StocksTrades, "*"); err != nil {
-		log.Error(err)
-	}
-
-	time.Sleep(250 * time.Millisecond)
-	if err := c.Unsubscribe(polygonws.StocksTrades, "*"); err != nil {
-		log.Error(err)
-	}
-
-	time.Sleep(5 * time.Second)
-	if err := c.Unsubscribe(polygonws.StocksSecAggs, "MSFT"); err != nil {
-		log.Error(err)
-	}
-
-	for {
-		select {
-		case <-ctx.Done():
-			return
-		}
-	}
+func TestSupportsTopic(t *testing.T) {
+	assert.Equal(t, true, supportsTopic("stocks", Topic(12)))
```

Comment:
can we use the actual topic var instead of casting an int to Topic? this isn't very readable as is